src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,31 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animalName) {
+    if (animalName instanceof Herbivore && animalName.hidden === false) {
```

Comment:
The condition `animalName instanceof Herbivore` might not be the intended logic. If `Carnivore` should be able to bite any `Animal`, consider changing this condition to `animalName instanceof Animal`. This would allow `Carnivore` to bite any animal, not just `Herbivore`.